src/ParallelAlgorithms/Events/ObserveFields.hpp

```diff
@@ -124,7 +132,18 @@ class ObserveFields<VolumeDim, ObservationValueTag, tmpl::list<Tensors...>,
     }
   };
 
-  using options = tmpl::list<SubfileName, VariablesToObserve>;
+  struct InterpolateToMesh {
+    using type = Options::Auto<Mesh<VolumeDim>, Options::AutoLabel::None>;
+    static constexpr Options::String help =
+        "An optional mesh to which the variables are interpolated. This mesh "
+        "specifies any number of collocation points, basis, and quadrature on "
+        "which the observed quantities are evaluated. If no mesh is given, the "
+        "results will be evaluated on the default mesh the simulation is run "
+        "on.";
+  };
```

Comment:
I think we should make sure the user understand that if they specify this option, the _original_ data will be lost and only the interpolated data is written to disk. We can do these things:

- Add a `static type suggested_value() noexcept { return {}; }` so the executable prints a note on startup that this option differs from the suggestion. I think this would be sufficient warning to the user.
- Add a note to this help string that the user can add the event multiple times, with and without the interpolating mesh, if they want both the original and the interpolated data written out, possibly at different triggers.

Reply:
I'm with @wthrowe. Observing data is opt-in and if you choose to write data to a different grid then nobody should be surprised that they got data on the grid they requested. If someone is surprised then I think it's a lesson they should learn, even if it's learned the hard way once. I really don't need the code yelling at me all the time "Hey, you aren't observing on the same grid you're solving the PDEs on!" It's not like I get incorrect answers if I observe on a different grid. `suggested_values` should only be used in cases where if you don't use the suggested value the simulation is likely to fail or behave incorrectly. Anything else, make people choose what they want. Furthermore, if you interpolate to a mesh with the same or higher-degree basis you haven't actually lost anything (unless you interpolate across a discontinuity but that's a different problem completely).

Adding a note about having multiple events: I don't think we need to start documenting options of events for how events work. Having documentation on the class/event itself that it can be added more than once is okay, but I think it would make more sense to have a tutorial on events. Additionally, at least in the evolution code the observation example input file does several observations at once.

Reply:
Ok works for me